### PR TITLE
Integration tests for newly added tickets stream

### PR DESCRIPTION
**Pull request by sgandhi1311**
# Description of change

- Add the get, create, and update functionality in client.py
- Update base.py and test cases

**Note** - Dependent on #https://github.com/stitchdata/environments/pull/764

# Manual QA steps
 - All integration tests are passed on dev-vm.
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/client.py [hunk 1]
```diff
@@ -950,6 +990,27 @@ def create_deals(self):
...
+        record_uuid = str(uuid.uuid4()).replace('-', '')
+        data = {
+        "properties": {
+            "content": f"Created for testing purpose - {record_uuid}",
+            "hs_pipeline": "0",
+            "hs_pipeline_stage": "1",
+            "hs_ticket_priority": "MEDIUM",
+            "subject": f"Sample ticket name - {record_uuid}"
+        }
+        }
```
Comment (kethan1122) [inline tests/client.py:1007]:
```suggestion
        data = {
            "properties": {
                "content": f"Created for testing purpose - {record_uuid}",
                "hs_pipeline": "0",
                "hs_pipeline_stage": "1",
                "hs_ticket_priority": "MEDIUM",
                "subject": f"Sample ticket name - {record_uuid}"
            }
        }
```

File: tests/client.py [hunk 2]
```diff
@@ -950,6 +990,27 @@ def create_deals(self):
...
+
+        # generate a record
+        response = self.post(url, data)
+        records = [response]
+        return records
```
Comment (kethan1122) [inline tests/client.py:1012]:
```suggestion
        return [response]
```
please remove the inline variable which is immediately returned

File: tests/client.py [hunk 3]
```diff
@@ -1465,6 +1528,27 @@ def update_engagements(self, engagement_id):
...
+            "properties": {
+                "subject": f"update record for testing - {record_uuid}"
+            }
+        }
+
+        self.patch(url, data)
+
+        record = self._get_tickets_by_pk(ticket_id)
+
+        return record
```
Comment (kethan1122) [inline tests/client.py:1550]:
```suggestion
        return self._get_tickets_by_pk(ticket_id)
```

File: tests/client.py [hunk 4]
```diff
@@ -1465,6 +1528,27 @@ def update_engagements(self, engagement_id):
...
+        Hubspot API https://developers.hubspot.com/docs/api/crm/tickets
+        :params ticket_id: the pk value of the ticket record to update
+        :return:
+        """
+        url = f"{BASE_URL}/crm/v4/objects/tickets/{ticket_id}"
+
+        record_uuid = str(uuid.uuid4()).replace('-', '')[:20]
+        data = {
+            "properties": {
+                "subject": f"update record for testing - {record_uuid}"
```
Comment (kethan1122) [inline tests/client.py:1542]: What's the purpose of sending this random `uuid` in the subject string ? Can't we use ticketID instead ?

File: tests/client.py [hunk 5]
```diff
@@ -655,6 +657,42 @@ def get_workflows(self):
...
+                    record
+                    for record in response["results"]
+                    if record[replication_key] >= self.start_date_strf.replace('.Z','.000Z')
+                ]
+            )
+
+            if not response.get("paging"):
+                break
+            params["after"] = response.get("paging").get("next").get("after")
+        return records
```
Comment (kethan1122) [inline tests/client.py:695]: A new empty line expected after end of function.

Comment (sgandhi1311) [reply tests/client.py:694]: done

File: tests/client.py (hunk 4 above)
Comment (sgandhi1311) [reply tests/client.py:1546]: This is to get the new value every time we update the record. TicketID will always be constant.

File: tests/client.py (hunk 3 above)
Comment (sgandhi1311) [reply tests/client.py]: done

File: tests/client.py (hunk 2 above)
Comment (sgandhi1311) [reply tests/client.py]: done

File: tests/client.py (hunk 1 above)
Comment (sgandhi1311) [reply tests/client.py]: done, reformated the whole file.

**Review by kethan1122 — APPROVED** [review]

**Review by rdeshmukh15 — APPROVED** [review]

File: tests/client.py [hunk 6]
```diff
@@ -655,6 +656,43 @@ def get_workflows(self):
...
+        while True:
+            response = self.get(url, params=params)
+
+            records.extend(
+                [
+                    record
+                    for record in response["results"]
+                    if record[replication_key] >= self.start_date_strf.replace('.Z', '.000Z')
+                ]
+            )
```
Comment (RushiT0122) [inline tests/client.py:689]:
```suggestion
            records.extend([record
                            for record in response["results"]
                            if record[replication_key] >= self.start_date_strf.replace('.Z', '.000Z')])
```

Comment (RushiT0122) [reply tests/client.py:689]: This code has mix coding styles but for list comprehession lets use above style like used elsewhere in the same file.

File: tests/test_hubspot_interrupted_sync.py [hunk 7]
```diff
@@ -38,6 +38,12 @@ def simulated_interruption(self, reference_state):
         new_state['bookmarks']['engagements']['lastUpdated'] = None
         new_state['bookmarks']['engagements']['current_sync_start'] = engagements_bookmark
 
+        tickets_bookmark = self.timedelta_formatted(
+            reference_state['bookmarks']['engagements']['lastUpdated'],
+            days=-1, str_format=self.BASIC_DATE_FORMAT
+        )
```
Comment (RushiT0122) [inline tests/test_hubspot_interrupted_sync.py:44]:
```suggestion
        tickets_bookmark = self.timedelta_formatted(
            reference_state['bookmarks']['engagements']['lastUpdated'],
            days=-1, str_format=self.BASIC_DATE_FORMAT)
```

Comment (RushiT0122) [reply tests/test_hubspot_interrupted_sync.py:44]: Also why are we handlling `tickets` stream separately?

**Review by RushiT0122 — CHANGES_REQUESTED** [review]

File: tests/client.py (hunk 6 above)
Comment (sgandhi1311) [reply tests/client.py]: done

File: tests/test_hubspot_interrupted_sync.py (hunk 7 above)
Comment (sgandhi1311) [reply tests/test_hubspot_interrupted_sync.py]: we setting the bookmark value for tickets stream in this way. Followed the existing pattern.

**Review by RushiT0122 — APPROVED** [review]

**Pull request merged**